### PR TITLE
Fix/tri 522 job attribute last modified on never updated

### PR DESCRIPTION
**Pull request by ds-ext-ooyeneye**

### CODE REVIEW

File: irs-api/src/main/java/net/catenax/irs/connector/job/MultiTransferJob.java [hunk 1]
```diff
@@ -115,8 +115,14 @@ public static class MultiTransferJobBuilder {
...
         /* package */ MultiTransferJobBuilder transitionComplete() {
-            return transition(JobState.COMPLETED, JobState.TRANSFERS_FINISHED, JobState.INITIAL).job(
-                    job.toBuilder().jobCompleted(ZonedDateTime.now(ZoneOffset.UTC)).build());
+            return transition(JobState.COMPLETED, JobState.TRANSFERS_FINISHED, JobState.INITIAL).job(job.toBuilder()
+                                                                                                        .jobCompleted(
+                                                                                                                ZonedDateTime.now(
+                                                                                                                        ZoneOffset.UTC))
+                                                                                                        .lastModifiedOn(
+                                                                                                                ZonedDateTime.now(
+                                                                                                                        ZoneOffset.UTC))
```
Comment (ds-jkreutzfeld) [inline irs-api/src/main/java/net/catenax/irs/connector/job/MultiTransferJob.java:124]: Why do this here if the transition method is called anyway?

File: irs-api/src/test/java/net/catenax/irs/connector/job/PersistentJobStoreTest.java [hunk 2]
```diff
@@ -95,7 +95,7 @@ void findByProcessId_WhenNotFound() {
     void create_and_find() {
         sut.create(job);
         assertThat(sut.find(job.getJobIdString())).isPresent();
-        assertThat(sut.find(otherJobId)).isEmpty();
```
Comment (ds-jkreutzfeld) [inline irs-api/src/test/java/net/catenax/irs/connector/job/PersistentJobStoreTest.java:98]: Why was this removed?

File: irs-api/src/test/java/net/catenax/irs/services/IrsItemGraphQueryServiceSpringBootTest.java [hunk 3]
```diff
@@ -99,6 +99,10 @@ void cancelJobById() {
 
         final JobState state = fetchedJob.get().getJob().getJobState();
         assertThat(state).isEqualTo(JobState.CANCELED);
+
+        final ZonedDateTime lastModifiedOn = fetchedJob.get().getJob().getLastModifiedOn();
+        assertThat(lastModifiedOn).isNotNull();
+        assertThat(lastModifiedOn).isBefore(ZonedDateTime.now());
```
Comment (ds-jkreutzfeld) [inline irs-api/src/test/java/net/catenax/irs/services/IrsItemGraphQueryServiceSpringBootTest.java:105]: This assertion doesn't make much sense to me.

Also in none of the tests there is an assertion that the lastModifiedDate has been updated over the lifetime of a job, please add that.

**Review by ds-jkreutzfeld — CHANGES_REQUESTED** [review]

File: irs-api/src/main/java/net/catenax/irs/connector/job/MultiTransferJob.java (hunk 1 above)
Comment (ds-ext-ooyeneye) [reply irs-api/src/main/java/net/catenax/irs/connector/job/MultiTransferJob.java:124]: Removed

File: irs-api/src/test/java/net/catenax/irs/connector/job/PersistentJobStoreTest.java (hunk 2 above)
Comment (ds-ext-ooyeneye) [reply irs-api/src/test/java/net/catenax/irs/connector/job/PersistentJobStoreTest.java:98]: Was done by Jaro

File: irs-api/src/test/java/net/catenax/irs/services/IrsItemGraphQueryServiceSpringBootTest.java (hunk 3 above)
Comment (ds-ext-ooyeneye) [reply irs-api/src/test/java/net/catenax/irs/services/IrsItemGraphQueryServiceSpringBootTest.java:105]: Another test rewritten for this test in PersistentJobStoreTest and InMemoryJobTest

File: irs-api/src/test/java/net/catenax/irs/connector/job/PersistentJobStoreTest.java (hunk 2 above)
Comment (ds-jhartmann) [reply irs-api/src/test/java/net/catenax/irs/connector/job/PersistentJobStoreTest.java:98]: No it wasn't.

Comment (ds-ext-ooyeneye) [reply irs-api/src/test/java/net/catenax/irs/connector/job/PersistentJobStoreTest.java]: Assertion statement reinstated

**Review by ds-jkreutzfeld — APPROVED** [review]

**Pull request merged**